### PR TITLE
Issue #46: Download, extract and format LibriVox data

### PR DESCRIPTION
**Pull request by reuben**
@kdavis-mozilla can you review this?

### CODE REVIEW

**Comment by reuben** [comment]
Hm, unless I'm missing something, this also fixes issue #47, it's just about switching the import line and the data_dir parameter and it should work just like TED-LIUM.

**Comment by kdavis-mozilla** [comment]
I'm going to evaluate this relative to pull request #50 as the majority of changes here are from there.

**Comment by kdavis-mozilla** [comment]
Further comment to get things past the download, extract, convert steps I had to run the code more than once.

File: util/importers/librivox.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
+import fnmatch
+import numpy as np
+import os
+import random
+import tarfile
+
+from glob import glob
+from itertools import cycle
+from math import ceil
+from pydub import AudioSegment
```
Comment (reuben) [inline util/importers/librivox.py]: Fixed.

**Comment by kdavis-mozilla** [comment]
Did you test this?

File: util/importers/librivox.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+from queue import PriorityQueue
```
Comment (reuben) [inline util/importers/librivox.py]: Fixed.

**Comment by reuben** [comment]
I'm running it on the entire dataset right now. Will report when it's completed.

**Comment by reuben** [comment]
(I did test on just dev-clean, and it worked.)

**Comment by kdavis-mozilla** [comment]
OK. In the process of the review I'll try on the entire train set. I'll run it overnight. I'm guessing it'll take about 5 hours.

**Comment by reuben** [comment]
OK everything should be fine now. Some files ended up with a \n and some didn't.

**Comment by kdavis-mozilla** [comment]
Data munging is always fun. Not!

**Comment by kdavis-mozilla** [comment]
It failed last night with this error
```
Traceback (most recent call last):
  File "DeepSpeech.py", line 295, in <module>
    train()
  File "DeepSpeech.py", line 230, in train
    librivox = read_data_sets(tf.get_default_graph(), '/data/OpenSLR', batch_size, n_input, n_context)
  File "/home/kdavis/Code/DeepSpeech/util/importers/librivox.py", line 176, in read_data_sets
    train = _read_data_set(graph, work_dir, "train-*-wav", thread_count, batch_size, numcep, numcontext)
  File "/home/kdavis/Code/DeepSpeech/util/importers/librivox.py", line 249, in _read_data_set
    return DataSet(graph, txt_files, thread_count, batch_size, numcep, numcontext)
  File "/home/kdavis/Code/DeepSpeech/util/importers/librivox.py", line 49, in __init__
    self._files_circular_list = self._create_files_circular_list()
  File "/home/kdavis/Code/DeepSpeech/util/importers/librivox.py", line 66, in _create_files_circular_list
    wav_file_size = os.path.getsize(wav_file)
  File "/usr/local/tensorflow-env/lib/python2.7/genericpath.py", line 49, in getsize
    return os.stat(filename).st_size
OSError: [Errno 2] No such file or directory: '/data/OpenSLR/LibriSpeech/train-clean-100-wav/289-121665-0042.wav'
```

I haven't tries to run it again to see if it works the second time.

However one thing of note is that
```
/data/OpenSLR/LibriSpeech/train-clean-100-wav/289-121665-0042.txt
```
exists.

**Comment by kdavis-mozilla** [comment]
So it seems as if there is still some problem in the conversion code.

**Comment by reuben** [comment]
I switched to SoX and tested it on the dev-clean and train-clean-100 datasets. @kdavis-mozilla, can you remove the extracted and converted files before you try this again? Just to make sure there's no weird state leftover from previous runs interfering with things.

Something like:

    rm -rf data/OpenSRL/LibriSpeech/{train,dev,test}-*

**Comment by kdavis-mozilla** [comment]
@reuben I'll remove any old state before re-testing. (In the past I've removed the entire `LibriSpeech` directory)

**Comment by kdavis-mozilla** [comment]
This looks good to me. I'm merge this with the two other pull requests.

**Comment by kdavis-mozilla** [comment]
Merged with pull request #49

**Pull request closed without merging**